src/sudoku.rs

```diff
@@ -237,8 +237,12 @@ impl<const N: usize> Board<N> {
 impl<const N: usize> Display for Board<N> {
     #[inline]
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        for row in &self.0 {
-            writeln!(f, "{}", row)?;
+        for (i, row) in (&self.0).iter().enumerate() {
```

Comment:
Minor change in how the Board is displayed - no longer prints a newline for the last line since the caller will probably add the newline.